src/cluster_legacy.c

```diff
@@ -2925,6 +3019,23 @@ int clusterProcessPacket(clusterLink *link) {
     clusterMsg *hdr = (clusterMsg *)link->rcvbuf;
     uint16_t type = ntohs(hdr->type);
     mstime_t now = mstime();
+    int is_light = IS_LIGHT_MESSAGE(type);
+
+    if (is_light) {
+        type &= ~CLUSTERMSG_MSB;
```

Comment:
consider adding an inline function to mask off these "modifier" bits. then we just need to do it once on [L2902](https://github.com/valkey-io/valkey/pull/654/files#diff-2515500619600c5a1e7a8d9aaa8761a6071314cd226c3b7ee4d0e054691f0a8eR2902)

```c
    uint16_t type = clusterGetMessageType(ntohs(hdr->type));
```

Reply:
We would not know if its light type, If we mask off before we determine if its a light message

Reply:
you can get the "modifier bits" too at that time. I think it will be cleaner to centralize all type related parsing logic in one place as opposed to each message type writing its own parsing logic. An alternative would be always referencing types and these "modifiers" via helper functions (such as `getType`, `isLight`, `isXXX`, etc) and this way we have consistency in how these message types are decoded.

Reply:
Oh I get it

Reply:
Done